aten/src/TH/generic/THTensorMoreMath.cpp

```diff
@@ -42,6 +40,16 @@ int THTensor_(equal)(THTensor *ta, THTensor* tb)
   return equal;
 }
 
+int THTensor_(equal)(THTensor *ta, THTensor* tb) {
```

Comment:
it looks like this is only called from functions like `bool _th_equal` in LegacyTHFunctionsCPU.cpp. Would it be easy to return a bool here (and in the cuda version) instead of returning an int and casting farther up the stack?

Reply:
The original function returned an `int` so I am following that convention. I don't know the answer to whether or not a bool return here changes the behavior.